### PR TITLE
Release artefacts

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: .github/workflows/builder.yaml [hunk 1]
```diff
@@ -25,3 +30,14 @@ jobs:
       uses: actions/checkout@v3
     - name: Build
       run: ./build.sh
+    - name: Tag
+      uses: rickstaa/action-create-tag@v1
+      with:
+        tag: ${{ steps.build.outputs.release }}
+  tidy:
```
Comment (eregon) [inline .github/workflows/builder.yaml:37]: You'd want to tidy only after both jobs above ran, so `needs: [build-debug, build-release]`

**Comment by chrisseaton** [comment]
I can't believe how wildly complicated it is to create a release from GitHub Actions. Seems like it should be a one-liner!

Anyhow this PR does make the builds available, via normal GitHub Actions artefacts. Just a shame they're double-zipped. I might come back to doing a proper GitHub Release later.

**Comment by eregon** [comment]
Actually a release is just `actions/upload-release-asset@v1` + `actions/create-release@v1` (or the newer action), so shouldn't be very different if at all.

True just a GHA artifact is easier, then you don't even need a tag and can just have a daily workflow.

**Comment by chrisseaton** [comment]
I don't know - I spent hours last night trying and it's a maze of unmaintained code, code that has assumptions about how they're going to be used that doesn't make any sense to me, and the complexity of this https://github.com/ruby/ruby-dev-builder/blob/master/.github/workflows/build.yml.

Maybe I should write my own Action that just takes a set of files and does the tagging, the uploading, the removing old versions.

Can't see why such an obvious thing for all projects (nightly builds) is so hard...

**Comment by eregon** [comment]
I think it's not so complex, most of that file is building+testing CRuby which can be ignored.
If you want to build on N platforms, i.e., ship binaries, then you need multiple build jobs and "merge" after, and so probably the same or similar complexity as that setup.
If you don't need anything platform-specific, then do everything in a single job, and maybe use https://github.com/softprops/action-gh-release.

If you don't care about a stable URL to get latest though I think artifacts on a daily workflow is easiest.

**Pull request merged**